File: ChatApp/app/src/main/java/top/chilfish/chatapp/adapter/ChatListAdapter.java
package top.chilfish.chatapp.adapter;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;

import java.util.List;

import top.chilfish.chatapp.R;
import top.chilfish.chatapp.databinding.ItemChatBinding;
import top.chilfish.chatapp.entity.ChatItem;

public class ChatListAdapter extends BaseAdapter<ChatItem> {
  private List<ChatItem> mChatList;

  public ChatListAdapter(List<ChatItem> data) {
    super(data);
    mChatList = data;
  }

  @NonNull
  @Override
  public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
    ItemChatBinding binging = ItemChatBinding.inflate(LayoutInflater.from(parent.getContext()),
        parent, false);

    return new ChatListViewHolder(binging);
  }

  public class ChatListViewHolder extends BaseAdapter<ChatItem>.ViewHolder {
    private ItemChatBinding binding;

    public ChatListViewHolder(ItemChatBinding binding) {
      super(binding.getRoot());
      this.binding = binding;
    }

    @Override
    public void bindData(ChatItem data) {
      try {
        Glide.with(itemView.getContext())
            .load(data.getProfile().getAvatar())
            .into(binding.chatAvatar);

        binding.chatName.setText(data.getProfile().getName());
        binding.chatContent.setText(data.getContent());
        binding.chatTime.setText(data.getTime());
      } catch (Exception e) {
        e.printStackTrace();
      }
    }
  }
}

File: ChatApp/app/src/main/java/top/chilfish/chatapp/adapter/ChatListAdapter.kt
package top.chilfish.chatapp.adapter

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.core.content.ContextCompat.startActivity
import com.bumptech.glide.Glide
import top.chilfish.chatapp.R
import top.chilfish.chatapp.databinding.ItemChatBinding
import top.chilfish.chatapp.entity.ChatItem
import top.chilfish.chatapp.ui.activities.ChatMainActivity

class ChatListAdapter : BaseAdapter<ChatItem, ItemChatBinding>() {
    override fun getLayoutId(): Int {
        return R.layout.item_chat
    }

    override fun onBindViewHolder(holder: ViewHolder<ItemChatBinding>, position: Int) {
        super.onBindViewHolder(holder, position)
        val data = items[position]
        val binding = holder.binding
        try {
            Glide.with(holder.itemView.context)
                .load(data.profile.avatar)
                .into(binding.chatAvatar)

            binding.chatName.text = data.profile.name
            binding.chatContent.text = data.content
            binding.chatTime.text = data.time
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun onItemClicked(item: ChatItem, view: View) {
        val bundle = Bundle()
        bundle.putSerializable("profile", item.profile)
        val intent = Intent(view.context, ChatMainActivity::class.java)
        intent.putExtras(bundle)
        startActivity(view.context, intent, null)
    }
}